src/cljx/schema/core.cljx

```diff
@@ -473,47 +422,71 @@
 
 (clojure.core/defrecord Either [schemas]
   Schema
-  (walker [this]
-          (let [sub-walkers (mapv subschema-walker schemas)]
-            (clojure.core/fn [x]
-              (loop [sub-walkers (seq sub-walkers)]
-                (if-not sub-walkers
-                  (macros/validation-error
-                   this x
-                   (list 'some (list 'check '% (utils/value-name x)) 'schemas))
-                  (let [res ((first sub-walkers) x)]
-                    (if-not (utils/error? res)
-                      res
-                      (recur (next sub-walkers)))))))))
+  (spec [this]
+    (variant/variant-spec
+     spec/+no-precondition+
+     (for [s schemas]
+       {:guard (complement (checker s))
```

Comment:
why is this guard here, i thought having guard would be equivalent? is it necessary for other applications besides checking?

seems like it is at least worth adding a small comment.